freva-rest/src/databrowser_api/core.py

```diff
@@ -762,13 +795,11 @@ def _get_url(self) -> tuple[str, Dict[str, Any]]:
         url = f"{self._config.get_core_url(core)}/select/"
         query = []
         for key, value in self.facets.items():
-            if key in self.uniq_keys:
-                search_value = " OR ".join(map(str, value))
-            else:
-                search_value = " OR ".join(map(str.lower, value))
-            for char in self.escape_chars:
-                search_value = search_value.replace(char, "\\" + char)
-            query.append(f"{key.lower()}:({search_value})")
+            query_pos, query_neg = self._join_facet_queries(key, value)
+            if query_pos:
+                query.append(f"{key.lower()}:({query_pos})")
+            if query_neg:
+                query.append(f"-{key.lower()}:({query_neg})")
```

Comment:
I do not get it, if the negation is done in https://github.com/FREVA-CLINT/freva-nextgen/blob/668fcc086d72322d3047254819266b18f43d2025/freva-rest/src/databrowser_api/core.py#L777
what happens if you do it again here
```python
            if query_neg:
                query.append(f"-{key.lower()}:({query_neg})")
```

I added to query_neg so it would strip either not, NOT, ! or - 
and then add that minus to the query itself

Reply:
We are not doing it again. My idea was to collect negative search statements (searches that should NOT contain the search result) and positive ones (facets that should contain search results). Once we have those we need to translate this to lucene syntax. Unfortunately, this doesn't work in lucene:

`{"fq": "project":"-cmip6"} ` it needs to be `{"fq":"-project":"cmip6"}`.

I am not super happy about this but there is nothing we can do.

In my opinion, something like project=-cmip6 is easy to understand and it is easier to implement than putting the not condition to the search key, because this: -project=cmip6 doesn't work in python nor at the cli. I guess this is why the old system had the` _not_ `construct. project_not_ = cmip.


We can reintroduce this for the sake of backward compatibility. But then we'd have to change the facet checker. Might not be a big deal though? Any opnios @MoSHad91 @eelucio